modules/mbedtls/Kconfig.tls-generic

```diff
@@ -567,6 +568,37 @@ config MBEDTLS_PSA_P256M_DRIVER_RAW
 	  Warning: Usage of this Kconfig option is prohibited in Zephyr's codebase.
 	           Users can enable it in case of very memory-constrained devices, but be aware that the p256-m interface is absolutely not guaranted to remain stable over time.
 
+config MBEDTLS_PSA_STATIC_KEY_SLOTS
+	bool "Use statically allocated key buffers to store key material"
+	default y if !MBEDTLS_ENABLE_HEAP
+	help
+	  By default Mbed TLS's PSA Crypto core uses heap memory to store the
+	  key's material for each key slot. This might impose an undesired
+	  requirement to support heap memory and its management code affecting
+	  RAM and ROM footprints at the same time.
+	  Enabling this symbol cause Mbed TLS to pre-allocate all the key slot
+	  buffers that are used to store the key's material at built time, thus
+	  removing the need for heap memory. Each buffer will be sized to
+	  contain the largest asymmetric/symmetric key type enabled in the build
+	  through PSA_WANT symbols.
+
+config MBEDTLS_PSA_KEY_SLOT_COUNT
+	int "Number of key slots in PSA Crypto core"
+	default 16
```

Comment:
What was the rationale for choosing 16? I'm wondering if we should make that default still a bit smaller.

Reply:
@frkv suggested this so I'll leave the answer to him

Reply:
I think that it will be very difficult to dictate a "one number that fits all" here, unless we start using `depends on` rules taking into consider the possibility of multiple supported protocols, number of consecutive links etc... 

The default number of keys allocated in the current version of Mbed TLS is 32, and we have already reduced this number to half of that.::

Please see:
https://github.com/Mbed-TLS/mbedtls/blob/107ea89daaefb9867ea9121002fbbdf926780e98/include/mbedtls/mbedtls_config.h#L4070

https://github.com/Mbed-TLS/mbedtls/blob/107ea89daaefb9867ea9121002fbbdf926780e98/include/psa/crypto_extra.h#L32

It is hard to answer definitely how many keys are needed at the same time, but if you set the default too low it may be a cause of runtime-failures that new user might waste precious time on...

If the end-product doesn't rely on that many keys, then it is adjustable down at will, when doing optimization...

Reply:
I was suggesting to set it down to ~12 because from my quick calculation (based on Valerio's showcase data in the PR description) it seemed that a number bigger than that would increase the RAM usage. Although in practice it actually depends on the key types enabled. So yeah, it's not that easy of a choice.

I was just trying to avoid increasing the footprint (because that goes against the original goal).
Maybe we could at least make sure to properly document this as a possible optimization path (so that users come across it easily).